### PR TITLE
[Silabs] [SiWx917] Added the platform chip_crypto to SiWx917 platform

### PR DESCRIPTION
**Pull request by chirag-silabs**
Added the tinycrypt in chip crypto to the platform specific
Adding the required files for the CHIPCryptoPal* 

Tested commissioning locally

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]
Rubber stamping, build files are ok. I did a quick overview of the crypto files. They are the tinycrypt implementation right?

The CLA part of the pull approve seem t to have failesld. You will probably have to rebase or make a bother plus to the pr to force a rerun

**Comment by jmartinez-silabs** [comment]
@rcasallas-silabs Can you take a look at this one. The CHIPCryptoPal file mostly

**Comment by chirag-silabs** [comment]
> mostly

Actually need to update this a bit, due to the changes done on this PR https://github.com/project-chip/connectedhomeip/pull/23792
Will update with the AES key modifications

File: src/platform/silabs/SiWx917/CHIPCryptoPAL.h [hunk 1]
```diff
@@ -0,0 +1,1703 @@
...
+
+    /**
+     * @brief Constructs span from the underlying buffer
+     */
+    ByteSpan Span() const { return ByteSpan(ConstBytes(), Length()); }
+
+    /**
+     * @brief Returns capacity of the buffer
+     */
+    static constexpr size_t Capacity() { return kCapacity; }
```
Comment (rcasallas-silabs) [inline src/platform/silabs/SiWx917/CHIPCryptoPAL.h:293]: This header duplicates src/crypto/CHIPCryptoPAL.h almost entirely. Generic changes like the new method Capacity() can be incorporated into the common code, and the rest should be added in a platform-specific header extending the common classes.

**Review by rcasallas-silabs — CHANGES_REQUESTED** [review]
Add common changes to `src/crypto/CHIPCryptoPAL.h`, and extend platform-specific features from the common code.

File: src/platform/silabs/SiWx917/CHIPCryptoPAL.h (hunk 1 above)
Comment (chirag-silabs) [reply src/platform/silabs/SiWx917/CHIPCryptoPAL.h]: Ohh okay, removed on the latest commit. Using the src/crypto/CHIPCryptoPAL.h directly

Comment (chirag-silabs) [reply src/platform/silabs/SiWx917/CHIPCryptoPAL.h]: @rcasallas-silabs modified based on the comments on the latest commit

**Review by rcasallas-silabs — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**